tests/integration/Mailer/WordPress/WordpressMailerTest.php

```diff
@@ -268,6 +275,131 @@ public function testItAddSubscriberMetaInfo() {
     $wpMailer->send();
   }
 
+  /*
+   * Test for issue https://mailpoet.atlassian.net/browse/MAILPOET-3707
+   */
+  public function testItPreservesWPReplyTo() {
+    list($mailer, $fallbackMailer) = $this->getMailerInstancesForReplyToTests();
+
+    $wpMailer = new WordPressMailer($mailer, $fallbackMailer, new MetaInfo, $this->subscribersRepository);
+    $wpMailer->addAddress('email@example.com', 'Full Name');
+    $wpMailer->addReplyTo('reply-to@example.com', 'Reply To');
+    $wpMailer->From = 'email-from@example.com';
+    $wpMailer->Body = 'Body';
+    $wpMailer->send();
```

Comment:
All the tests that I created here are sending the e-mails as I didn't find a way to check the reply-to address of the e-mails without actually sending them. There might be a way to achieve this using PHPUnit mocks, but I couldn't find it on my first try, and I didn't want to spend more time on this task before getting some feedback.